src/app/data-planes/components/RuleEntryList.vue

```diff
@@ -3,131 +3,149 @@
     :initially-open="[]"
     multiple-open
   >
-    <AccordionItem
-      v-for="(ruleEntry, index) in props.ruleEntries"
-      :key="index"
+    <template
+      v-for="policies in [props.rules.reduce<Record<string, Rule[]>>((prev, item) => {
+        if(typeof prev[item.type] === 'undefined') {
+          prev[item.type] = []
+        }
+        prev[item.type].push(item)
+        return prev
+      }, {})]"
+      :key="policies"
     >
-      <template #accordion-header>
-        <h3 class="policy-type-heading">
-          <PolicyTypeTag :policy-type="ruleEntry.type">
-            {{ ruleEntry.type }}
-          </PolicyTypeTag>
-        </h3>
-      </template>
-
-      <template #accordion-content>
-        <div class="policy-list">
-          <KTable
-            class="policy-type-table"
-            :class="{
-              'has-matchers': props.showMatchers,
-            }"
-            :fetcher="() => ({ data: ruleEntry.rules, total: ruleEntry.rules.length })"
-            :headers="[
-              ...(props.showMatchers ? [{ label: 'Matchers', key: 'matchers' }] : []),
-              { label: 'Origin policies', key: 'origins' },
-              { label: 'Conf', key: 'config' },
-            ]"
-            :cell-attrs="getCellAttributes"
-            disable-pagination
+      <AccordionItem
+        v-for="(items, type) in policies"
+        :key="type"
+      >
+        <template #accordion-header>
+          <h3 class="policy-type-heading">
+            <PolicyTypeTag :policy-type="type">
+              {{ type }}
+            </PolicyTypeTag>
+          </h3>
+        </template>
+
+        <template #accordion-content>
+          <template
+            v-for="hasMatchers in [items.some((item) => item.matchers.length > 0)]"
+            :key="hasMatchers"
           >
-            <template
-              v-if="props.showMatchers"
-              #matchers="{ row }: { row: RuleEntryRule }"
-            >
-              <span
-                v-if="row.matchers && row.matchers.length > 0"
-                class="matcher"
+            <div class="policy-list">
+              <AppCollection
+                class="policy-type-table"
+                :class="{
+                  'has-matchers': hasMatchers,
+                }"
+                :total="items.length"
+                :items="items"
+                :page-size="100000"
```

Comment:
Oh I nearly passed `Number.POSITIVE_INFINITY`.

Ummm, from what I remember was to turn off the pagination otherwise from what I follow the pagination will kick in at 30 items, and we don't want it to. `100000` felt like a sufficiently high number to avoid paging without using `Number.POSITIVE_INFINITY`

Reply:
I’d rather we use a more direct signal that indicates to AppCollection to not paginate whatsoever. Remembering to set a really high number to do that is just a weird interface.

Reply:
Yes agree, but similar to the `a` selector thing I didn't really want to start changing these deeper view things in a data-type PR.

We should probably say "if you don't define a page number then don't page".